src/libstd/future.rs

```diff
@@ -26,6 +26,7 @@ pub fn from_generator<T: Generator<Yield = ()>>(x: T) -> impl Future<Output = T:
 #[doc(hidden)]
 #[unstable(feature = "gen_future", issue = "50547")]
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+#[cfg_attr(not(bootstrap), rustc_diagnostic_item = "gen_future")]
```

Comment:
you don't need `bootstrap` cfgs for diagnostic items.

Reply:
:woman_facepalming: So I think the problem is that building libstd for running its tests will use libstd as its own dependency. as a quick fix can probably just 
```suggestion
#[cfg_attr(not(test), rustc_diagnostic_item = "gen_future")]
```

Though the fact that the `GenFuture` type gets compiled twice is probably already not great and maybe the entire type just needs a `#[cfg(not(test))]` (I think other types have this, because of similar testing problems)

Reply:
Thanks @oli-obk! I've just changed it to `not(test)` for now.